compiler/modules/types/list.bal

```diff
@@ -320,8 +316,12 @@ function listInhabited(Context cx, FixedLengthArray members, SemType rest, ListC
             }
         }
         SemType rd = diff(rest, nt.rest);
-        if !isEmpty(cx, rd) && listInhabited(cx, members, rd, neg.next) {
-            return true;
+        if !isEmpty(cx, rd) {
+            FixedLengthArray s = fixedArrayShallowCopy(members);
+            fixedArrayFill(s, maxInitialLen, rest);
```

Comment:
@jclark I think this is the logical equivalence to what we do for each list member, applied to the `rest`.

Reply:
What if `maxInitialLen < len`?

Reply:
updated.